### PR TITLE
fixed intel build

### PR DESCRIPTION
**Pull request by 1vipulgupta**

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -71,13 +71,16 @@
...
             just
             nixci
             # For when we start using Tauri
             cargo-tauri
             trunk
-          ];
+          ]) ++ lib.optionals pkgs.stdenv.isDarwin (
+            [ pkgs.darwin.apple_sdk_11_0.frameworks.IOKit
+              pkgs.darwin.apple_sdk_11_0.frameworks.Security
+            ]);
```
Comment (srid) [inline flake.nix:83]: This should go in rust.nix (`rustBuildInputs`). Also, CI fails due to not having run autoformatter.

**Review by srid — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (1vipulgupta) [reply flake.nix]: Adding in `rustBuildInputs` is causing nix run failing. Updated and added to `devShells.rust`

Comment (srid) [reply flake.nix]: > causing nix run failing

Please post the full error.

---

devShells.rust already uses rustBuildInputs transitively, so the devShell will now have duplicate IOKit and Security frameworks.

Comment (1vipulgupta) [reply flake.nix]: `error occurred: Command env -u IPHONEOS_DEPLOYMENT_TARGET "clang" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=x86_64-apple-darwin" "-Wall" "-Wextra" "-o" "/private/tmp/nix-build-nix-browser-deps-0.1.0.drv-0/source/target/release/build/tao-fde661b9405e71cb/out/src/platform_impl/macos/carbon_hotkey/carbon_hotkey_binding.o" "-c" "src/platform_impl/macos/carbon_hotkey/carbon_hotkey_binding.c" with args "clang" did not execute successfully (status code exit status: 1).`

**Comment by srid** [comment]
`main` branch builds on Intel mac (https://github.com/juspay/nix-browser/blob/main/.github/workflows/ci.yaml) and I've confirmed that the app runs locally in Rosetta.

**Pull request closed without merging**